File: frontend/types/definition.ts
/**
 * Term definition types for glossary/tooltip features.
 */

export type DefinitionLanguage = 'en' | 'ta' | 'both';

export interface TermDefinition {
  term_id: string;
  name_en?: string | null;
  name_ta?: string | null;
  short_definition_en?: string | null;
  short_definition_ta?: string | null;
  detailed_explanation_en?: string | null;
  detailed_explanation_ta?: string | null;
  significance_tradition?: string | null;
  calculation_method?: string | null;
  related_terms?: string[] | null;
  sources?: string[] | null;
  icon?: string | null;
}

export interface TermDefinitionsResponse {
  definitions: TermDefinition[];
  count: number;
  language: DefinitionLanguage;
}

File: backend/src/models/definition.py
"""
Term definition models for tooltip and glossary APIs.
"""

from typing import Literal, Optional

from pydantic import Field, field_validator

from .base import BaseModel


class TermDefinition(BaseModel):
    """Educational definition for a Panchang term."""

    term_id: str = Field(..., description="Unique term identifier (snake_case)")
    name_en: Optional[str] = Field(default=None, description="English term name")
    name_ta: Optional[str] = Field(default=None, description="Tamil term name")
    short_definition_en: Optional[str] = Field(default=None, description="Short English definition")
    short_definition_ta: Optional[str] = Field(default=None, description="Short Tamil definition")
    detailed_explanation_en: Optional[str] = Field(
        default=None, description="Detailed English explanation"
    )
    detailed_explanation_ta: Optional[str] = Field(
        default=None, description="Detailed Tamil explanation"
    )
    significance_tradition: Optional[str] = Field(
        default=None, description="Tradition-specific significance"
    )
    calculation_method: Optional[str] = Field(default=None, description="How the term is calculated")
    related_terms: Optional[list[str]] = Field(default=None, description="Related term IDs")
    sources: Optional[list[str]] = Field(default=None, description="Reference sources")
    icon: Optional[str] = Field(default=None, description="Icon identifier")

    @field_validator("term_id")
    @classmethod
    def validate_term_id(cls, value: str) -> str:
        if not value or not value.strip():
            raise ValueError("term_id cannot be empty")
        return value.strip()


class DefinitionsListResponse(BaseModel):
    """Response for list definitions endpoint."""

    definitions: list[TermDefinition]
    count: int
    language: Literal["en", "ta", "both"] = "both"
